src/Chapter4.hs

```diff
@@ -472,10 +485,11 @@ Implement the Applicative instance for our 'Secret' data type from before.
 -}
 instance Applicative (Secret e) where
     pure :: a -> Secret e a
-    pure = error "pure Secret: Not implemented!"
-
+    pure = Reward
     (<*>) :: Secret e (a -> b) -> Secret e a -> Secret e b
-    (<*>) = error "(<*>) Secret: Not implemented!"
+    (Trap e) <*> _ = Trap e
+    _ <*> (Trap e) = Trap e
+    (Reward a2b) <*> (Reward a) = Reward $ a2b a
```

Comment:
You can notice that these two cases remind `Functor` instance for `Secret` and you can reuse it here 😉